### ISSUE TITLE
[107] Why Can’t I Open My Drawer? Mitigating Object-Driven Shortcuts in Zero-Shot Compositional Action Recognition

### ISSUE THREAD

**Issue by anthony-kim-sb** [dhkim0225/1day_1paper#138]
data sparse 한 real world 문제를 효과적으로 풀려면 결국 zero-shot 성능이 어느정도는 나와줘야 하지 않나 싶다.
논문은 Zero-Shot Compositional Action Recognition (ZS-CAR) 문제를 푼다.

<img width="847" height="76" alt="Image" src="https://github.com/user-attachments/assets/d356a1bf-870d-4481-8d39-959b515f4f99" />

# INTRO

ZS-CAR 문제 풀 때는
action을 하나의 atomic class로 외우는 게 아니라, verb와 object라는 두 primitive를 따로 이해한 뒤 조합하는데
- (open, drawer)
- (close, window)
- (lift, spoon)
- (put, cup)

이게 맨날 drawer 는 close 하고 있는 데이터만 학습하면 open 한다가 학습이 안되버린다.
그니까, "open" 한다. 랑 drawer 를 좀 더 분리해 보자는 아이디어.
그래야 o-pen 하게 학습을 좀 더 잘 할 수 있다는 것이다.

기존 방법은 unseen generalization이 약한 게 아니라, 학습이 진행될수록 점점 더 shortcut에 과적합되고, 그래서 compositional reasoning의 이득 자체를 잃는다.

<img width="1346" height="499" alt="Image" src="https://github.com/user-attachments/assets/5c23ea46-b758-4d70-aea1-ef1746e0ec45" />

그래서 RCORE 를 제안할 건데, 2가지 특성을 지닌다.

1. composition-aware augmentation
  - data-level intervention
  - sparse training 공간을 넓혀 줘서 모델이 특정 co-occurrence에 덜 매달리게 만들려는 목적. 
  - 비디오의 temporal structure를 망가뜨리지 않아야 함. 그냥 아무 mixup을 해버리면 verb를 배우는 데 필요한 motion cue까지 깨질 수 있음.

2. temporal order regularization objective
  - learning objective-level intervention
  - static cue에 의존하지 않게 regularization
  - 기존 temporal representation과 temporal perturbation된 representation을 비교해서 verb 에 대한 결과가 다르도록 학습하겠다. 
  - 프레임 순서를 뒤집거나 섞었을 때도 verb prediction이 별로 안 흔들리면 그건 좋은 verb learner가 아니다.

그리고, Compositional video understanding의 주된 병목은 model capacity가 아니라 model behavior다.
모델을 키우면 shortcut이 더 강화될 뿐이다.

# 2.Related Works

### 2.1 Zero-shot compositional action recognition
일반 zero-shot action recognition 과는 다르다

- zero-shot action recognition 은 테스트에서 새 action class 자체가 등장함
- open-set action recognition은 훈련에 없던 class를 거부/구별하는 문제가 섞여 있음. 
- ZS-CAR은 verb class도 알고, object class도 안다. 다만 그 조합이 처음 보는 것. 즉 새로운 primitive를 맞히는 게 아니라, 익숙한 primitive를 새롭게 compose할 수 있는가를 보는 문제.

기존 ZS-CAR 계열 방법들은 크게 두 갈래.
- conditional learning: verb와 object의 결합 확률을 더 잘 모델링하려는 접근
- feature disentanglement: verb/object 성분을 더 잘 분리해서 표현하려는 접근

[C2C](https://arxiv.org/abs/2407.06113)는 conditional modeling으로 joint verb-object probability를 추정하는 강한 baseline이고, 다른 최근 연구는 [coding rate reduction](https://openaccess.thecvf.com/content/ICCV2025/papers/Jung_Zero-Shot_Compositional_Video_Learning_with_Coding_Rate_Reduction_ICCV_2025_paper.pdf)이나 disentanglement 쪽을 연구함. 

이런 방법들이 조합을 모델링한다는 표면적 목표는 갖고 있어도, 실제 학습 과정에서는 여전히 object-driven shortcut에 취약하다

### 2.2 Image compositional zero-shot learning

이미지 쪽 compositional zero-shot learning은 오래 연구됐고, 보통 아래와 같은 조합을 다룬다.
- object + state
- attribute + object

예를 들어 red car, old chair, sliced apple 같은 식. 

이미지 CZSL에서는 state - object 조합이 주로 정적인 패턴으로 드러나는데, video ZS-CAR에서는 verb가 시간축을 따라 정의된다.

## 3. Diagnosis: Why ZS-CAR Models Fail?

$Y^V$: verb label set 
$Y^O$: object label set 
$y=(y^V, y^O)\in Y^V \times Y^O$ : action label

학습 데이터는 다음과 같고

$$
D_{\text{train}}=\{(X_i,y_i)\}_{i=1}^N
$$

실제 training에서 관측된 pair들의 집합은 다음과 같고

$$
Y_{\text{seen}}=\{y_i \mid (X_i,y_i)\in D_{\text{train}}\}
$$

unseen composition space 는 다음과 같다

$$
Y_{\text{unseen}} = (Y^V \times Y^O)\setminus Y_{\text{seen}}
$$ 

primitive는 known이지만 조합이 unseen인 pair들의 집합.
validation/test는 seen과 unseen 조합을 둘 다 포함.
이 기준에서 compositional generalization을 평가.

즉, 이 문제는 class extrapolation이 아니라 combination extrapolation이다.

### 3.2 FSP & FCP
학습 중 bias를 측정하는 지표를 두 개를 정의한다.

False Seen Prediction (FSP)
정답이 unseen composition인 샘플에 대해, 모델이 틀리면서 seen composition label로 예측한 비율

$$
\text{FSP} = \frac{1}{|D^{\text{unseen}}_{\text{val}}|} \sum_i \mathbf{1}[\hat y_i \in Y_{\text{seen}}]
$$

False Co-occurrence Prediction (FCP)
FSP 중에서도 training에서 자주 같이 나왔던 pair로 예측하는 비율
training data에서의 conditional frequency
$p(o\mid v)$, $p(v\mid o)$ 가 각각 $\mu+\sigma$를 넘는 pair들의 교집합으로 $Y_{\text{freq}}$를 정의한다.

$$
\text{FCP} = \frac{\sum_i \mathbf{1}[\hat y_i \in Y_{\text{freq}}]} {\sum_i \mathbf{1}[\hat y_i \in Y_{\text{seen}}]}
$$

<img width="1338" height="541" alt="Image" src="https://github.com/user-attachments/assets/244fe104-93c8-4eb7-9527-7371e63b7a28" />

위 figure 를 해석하면,
a) object 는 verb 보다 더 쉽게 수렴한다.
b) [Sth-com](https://arxiv.org/abs/2407.06113)에서 4×4 toy set을 골라서 perfectly biased training set을 만들었다. 성능을 재보면 unseen 은 점점 줄어듦. 학습 할 때는 diagonal 애들만 보게 했다.

<img width="515" height="545" alt="Image" src="https://github.com/user-attachments/assets/66f37d9a-936e-4359-9b88-ab80dcb422db" />

### 3.2 Compositional Gap

$$
\Delta CG = Acc_C - (Acc_V \times Acc_O)
$$

- $\Delta CG > 0$: composition modeling이 실제로 추가 이득을 줌
- $\Delta CG \approx 0$: 사실상 component prediction 이상을 못 함
- $\Delta CG < 0$: joint composition head가 오히려 unseen pair에서 방해됨

<img width="660" height="386" alt="Image" src="https://github.com/user-attachments/assets/b0420419-fbe1-49ff-8d08-62ca1db9ea79" />

# 4. Robust COmpositional REpresentations (RCORE)

augmented video $\hat X$ 가 T개의 frame 을 갖고
backbone이 각 프레임에 대해 $f_t \in \mathbb{R}^D$ 형태의 frame-level feature를 낸다.
- verb feature sequence $F^V \in \mathbb{R}^{T \times D}$
- object feature $f^O \in \mathbb{R}^{D}$

### VOCAMix
희소한 조합 supervision
문제는 결국 sparse 한 점에서 오니까 aug 를 잘 해보자.
naive한 mixing은 오히려 verb를 더 망칠 수 있다. verb는 유지하고 object 만 바꾸자.

<img width="734" height="571" alt="Image" src="https://github.com/user-attachments/assets/7f7984c5-daef-4436-a4d8-80bcddaeb5ac" />

VOCAMix는 object를 완전히 바꾸는 게 아니라 \lambda 비율만큼 섞기 때문에, object label도 one-hot hard label이 아니라 soft label을 사용한다.

$$
\tilde y_i^O = (1-\lambda)y_i^O + \lambda y_j^O
$$

### TORC
verb/object 학습 난이도 비대칭 해결
verb representation이 object appearance가 아니라 temporal structure에 의존하도록 만들기.  ￼

frame-level feature sequence $F = (f_1,\dots,f_T)$ 가 있으면, 여기서 두 개의 temporal perturbation을 만든다.
- reversed sequence $F_{\text{rev}} = (f_T,\dots,f_1)$
- shuffled sequence $F_{\text{shuffled}} = \pi(F)$

여기서 $\pi$는 랜덤 permutation. 이들을 같은 verb encoder에 통과시켜서 feature 들을 만든다.
- $f^V_{\text{rev}}$
- $f^V_{\text{shuffled}}$

<img width="779" height="539" alt="Image" src="https://github.com/user-attachments/assets/368b1831-9e0d-4e1b-b18f-9e04cbe09879" />

Loss 의 첫 번째 항: $L_{\cos}$

$$
L_{\cos} = \cos(f^V, f^V_{\text{rev}})
$$

seq 를 반대로 했으니까 cos sim 방향이 달라야 한다는 말.
음... 이건 lambda 값이 크지 않아야 할 수도? reverse verb 라고 embedding 의 방향이 꼭 90도 꺾여야 할 필요는 없으니까?
그리고 사실 90도가 될 수 있는 방향이 너무 제각각이라서 데이터가 많지 않은 verb 는 아무튼 잘 모이기 힘들수도 있겠다.

두 번째 항: $L_{\text{ent}}$

temporal structure가 깨진 입력에 대해 confident verb prediction을 하지 못하게 만든다. 
negative entropy 형태의 loss를 정의

$$
L_{\text{ent}} = \sum_{i=1}^{|Y^V|} p_i \log p_i
$$

음.. 좋긴 한데, 이것도 너무 키우면 shortcut 이 강화될 것 같은데?

$$
L_{\text{TORC}} = L_{\cos} + L_{\text{ent}}
$$

아 뭔가 더 잘 설계할 수도 있을 것 같은데..

### Baseline loss

<img width="1083" height="584" alt="Image" src="https://github.com/user-attachments/assets/1e58b780-64e8-498f-b31c-7a9192c7bb93" />

1) Component loss: verb/object 각 branch에 대해 표준 cross-entropy를 쓴다.

$$
L_{\text{com}} = L_V + L_O
$$

visual feature와 corresponding text feature의 cosine similarity로 logits를 만들고, verb/object 예측을 따로 학습

2) Composition loss

C2C 방식. marginal scores $(p_V, p_O)$와 conditional logits $(p_{V|O}, p_{O|V})$를 결합해서 composition prediction을 만든다고 적는다.

$$
\hat y = \text{Softmax}(p_O p_{V|O} + p_V p_{O|V})
$$

$$
L_{\text{comp}} = L_{\text{CE}}(\hat y, y)
$$

- object를 어느 정도 자신 있게 알 때, 그 object 조건하에서 verb를 추론
- verb를 어느 정도 자신 있게 알 때, 그 verb 조건하에서 object를 추론
- 이 둘을 합쳐 pair prediction

### Margin loss
frequent co-occurrence prior dominance

frequent verb-object pair는 unseen에서도 너무 강한 distractor로 작동한다. 
ground truth가 아닌데도 logit이 높게 뜨는 frequent-but-incorrect composition을 직접 눌러주겠다

$$
L_M = \sum_{c \in C_{\text{margin}}} \max\big(0,\log \hat y(c)-\log \hat y(g)+m\big)
$$

$g$: ground-truth composition index
$C_{\text{margin}}$: top-K incorrect predictions와 frequently co-occurring compositions의 교집합

틀린 class 전체를 다 누르는 게 아니라 현재 sample에서 실제로 헷갈리는 hard negatives 중에서도, dataset bias에 의해 자주 튀는 녀석들만 선택해서 누른다. all-seen에 적용하는 것보다 top-K frequent incorrect에만 선택적으로 거는 게 더 좋다고 나온다. 

### 전체 objective

$$
L_{\text{total}} = \alpha L_{\text{com}} + \beta L_{\text{comp}} + \gamma L_{\text{TORC}} + \delta L_M
$$

# 5. Results
<img width="1357" height="474" alt="Image" src="https://github.com/user-attachments/assets/5cf181fa-4aee-45a5-925f-187879d0f96b" />

<img width="658" height="475" alt="Image" src="https://github.com/user-attachments/assets/41f185e5-7254-4129-9761-0d8c07b3fc30" />

<img width="1359" height="366" alt="Image" src="https://github.com/user-attachments/assets/94961598-1dd5-4377-8793-906028bec6ea" />

<img width="1359" height="511" alt="Image" src="https://github.com/user-attachments/assets/c050dfd8-9681-499e-8601-9b7e93961b2c" />
